crates/nu-derive-value/src/attributes.rs

```diff
@@ -0,0 +1,112 @@
+use convert_case::Case;
+use syn::{spanned::Spanned, Attribute, Fields, LitStr};
+
+use crate::{error::DeriveError, HELPER_ATTRIBUTE};
+
+#[derive(Debug)]
+pub struct ContainerAttributes {
+    pub rename_all: Case,
+}
+
+impl Default for ContainerAttributes {
+    fn default() -> Self {
+        Self {
+            rename_all: Case::Snake,
+        }
+    }
+}
+
+impl ContainerAttributes {
+    pub fn parse_attrs<'a, M>(
+        iter: impl Iterator<Item = &'a Attribute>,
+    ) -> Result<Self, DeriveError<M>> {
+        let mut container_attrs = ContainerAttributes::default();
+        for attr in filter(iter) {
+            // This is a container to allow returning derive errors inside the parse_nested_meta fn.
+            let mut err = Ok(());
+
+            attr.parse_nested_meta(|meta| {
+                let ident = meta.path.require_ident()?;
+                match ident.to_string().as_str() {
+                    "rename_all" => {
+                        let case: LitStr = meta.value()?.parse()?;
+                        let case = match case.value().as_str() {
```

Comment:
> I think we should try to follow the serde names. Maybe `lower` should be `lower with spaces`, `UPPER` be `UPPER WITH SPACES`, then the flat ones should be `lowercase` and `UPPERCASE`

I kept all the names I currently have as they are names of `convert_case`. But because all names are case sensitive and act as kind of a template, I added all `serde` names for the cases.

> I'm not sure what to call `Train-Case`, I've never heard it called this but agree it's useful since PowerShell uses it.

It's just the name that `convert_case` uses. In the generated documenation of the traits `IntoValue` and `FromValue` this is also referenced, so this should be fine.

> And I hate to be no-fun but we probably don't need tOGGLE and aLtErNaTiNg, I just can't imagine anyone wanting that in this context. 😛 Keep in mind that anything we add here, we're supporting as a public interface.

Yeah, not really useful. I just implemented all of them but now removed these two.